### PR TITLE
re-add polymer 1 rtl rules

### PR DESCRIPTION
**Pull request by margaree**
hopefully fixes https://github.com/BrightspaceUI/button/issues/121

I could only reproduce on a quad site with more-less within activity feed. @emil-furniss if you could locally verify in activity feed that this fixes the problem, that would be really helpful. Otherwise I'll try to get activity feed running locally or find another place to reproduce.

### CODE REVIEW

**Comment by awikkerink** [comment]
`:host-context` is weird. The polyfill and native behavior is inconsistent. The browser may also ignore an entire rule block if one of the css selectors is invalid.

File: d2l-button-icon.html [hunk 1]
```diff
@@ -37,6 +37,10 @@
 			:host([h-align="text"]) button {
 				left: calc(((var(--d2l-button-icon-min-width) - 0.9rem) / 2) * -1);
 			}
+			:host-context([dir="rtl"])[h-align="text"] button {
+				left: 0;
+				right: calc(((var(--d2l-button-icon-min-width) - 0.9rem) / 2) * -1);
+			}
```
Comment (awikkerink) [inline d2l-button-icon.html]: Maybe try:
```css
           :host-context([dir="rtl"])[h-align="text"] button {
				left: 0;
				right: calc(((var(--d2l-button-icon-min-width) - 0.9rem) / 2) * -1);
			}
			:host-context([dir="rtl"]):host([h-align="text"]) button {
				left: 0;
				right: calc(((var(--d2l-button-icon-min-width) - 0.9rem) / 2) * -1);
			}
```

The shadycss polyfill doesn't handle this case well.

File: d2l-button-icon.html [hunk 2]
```diff
@@ -37,6 +37,10 @@
...
 			:host(:dir(rtl))[h-align="text"] button {
```
Comment (awikkerink) [inline d2l-button-icon.html]: `:host(:dir(rtl))[h-align="text"]` -> `:host([h-align="text"]:dir(rtl))`

File: d2l-button-icon.html (hunk 1 above)
Comment (awikkerink) [reply d2l-button-icon.html]: Maybe use mixins to reduce code duplication?

**Comment by emil-furniss** [comment]
I can confirm that this fixes the problem in activity feed.

File: d2l-button-icon.html (hunk 2 above)
Comment (margaree) [reply d2l-button-icon.html]: so this line is for P2/shady and it doesn't seem to work if I switch it to the option on the right.

**Pull request merged**